.github/workflows/php-test-plugins.yml

```
@@ -68,6 +72,32 @@ jobs:
       - name: Composer Install
         run: npm run wp-env run tests-cli -- --env-cwd="wp-content/plugins/$(basename $(pwd))" composer install --no-interaction --no-progress
       - name: Running single site unit tests
-        run: npm run test-php
+        run: |
+          if [ "${{ matrix.coverage }}" == "true" ]; then
+            npm run test-php -- --coverage-clover=coverage.xml
+          else
+            npm run test-php
+          fi
       - name: Running multisite unit tests
-        run: npm run test-php-multisite
+        run: |
+          if [ "${{ matrix.coverage }}" == "true" ]; then
+            npm run test-php-multisite -- --coverage-clover=coverage-multisite.xml
+          else
+            npm run test-php-multisite
+          fi
+      - name: Upload single site coverage reports to Codecov
+        if: ${{ matrix.coverage == true }}
+        uses: codecov/codecov-action@v4
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: coverage.xml
+          flags: unittests
+          name: ${{ matrix.php }}-single-site-coverage
```

Let's make sure we're alerted of failures for coverage reporting:
```suggestion
          name: ${{ matrix.php }}-single-site-coverage
          fail_ci_if_error: true
```